The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12148415708/job/33876843016) [(plain)](https://github.com/rust-lang/rust/commit/b885c297074482979d8d093f3fd52b4166ffe86d/checks/33876843016/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 3.081 Building wheels for collected packages: reuse
#13 3.082   Building wheel for reuse (pyproject.toml): started
#13 3.335   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.336   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 3.336   Stored in directory: /tmp/pip-ephem-wheel-cache-7c2sdv9j/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.339 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.740 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.741 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.314 Collecting virtualenv
#13 4.314 Collecting virtualenv
#13 4.371   Downloading virtualenv-20.28.0-py3-none-any.whl (4.3 MB)
#13 4.455      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 52.8 MB/s eta 0:00:00
#13 4.517 Collecting platformdirs<5,>=3.9.1
#13 4.528   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.553 Collecting distlib<1,>=0.3.7
#13 4.572      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 81.4 MB/s eta 0:00:00
#13 4.614 Collecting filelock<4,>=3.12.2
#13 4.625   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.625   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.708 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.890 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.28.0
#13 DONE 5.0s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      317376 kB
DirectMap2M:     8071168 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2024-11-27/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 31.51s
##[endgroup]
fmt check
Diff in /checkout/src/librustdoc/html/sources.rs:66:
         let span = item.span(self.tcx);
         let Some(span) = span else { return };
         // skip all synthetic "files"
-        let Some(p) = filename_real_and_local(span, sess).and_then(|file| file.into_local_path()) else {
+        let Some(p) = filename_real_and_local(span, sess).and_then(|file| file.into_local_path())
             return;
         };
         };
         if self.local_sources.contains_key(&*p) {
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/src/librustdoc/clean/blanket_impl.rs" "/checkout/src/librustdoc/clean/inline.rs" "/checkout/src/librustdoc/clean/types/tests.rs" "/checkout/src/librustdoc/clean/utils/tests.rs" "/checkout/src/librustdoc/clean/cfg/tests.rs" "/checkout/src/librustdoc/clean/cfg.rs" "/checkout/src/librustdoc/clean/types.rs" "/checkout/src/librustdoc/clean/simplify.rs" "/checkout/src/librustdoc/clean/mod.rs" "/checkout/src/librustdoc/clean/render_macro_matchers.rs" "/checkout/src/librustdoc/html/static_files.rs" "/checkout/src/librustdoc/html/length_limit.rs" "/checkout/src/librustdoc/html/sources.rs" "/checkout/src/librustdoc/html/markdown/footnotes.rs" "/checkout/src/librustdoc/html/markdown/tests.rs" "/checkout/src/librustdoc/html/markdown.rs" "/checkout/src/librustdoc/html/url_parts_builder/tests.rs" "/checkout/src/librustdoc/html/toc.rs" "/checkout/src/librustdoc/html/layout.rs" "/checkout/src/librustdoc/html/url_parts_builder.rs" "/checkout/src/librustdoc/html/render/ordered_json/tests.rs" "/checkout/src/librustdoc/html/render/ordered_json.rs" "/checkout/src/librustdoc/html/render/search_index/encode.rs" "/checkout/src/librustdoc/html/render/type_layout.rs" "/checkout/src/librustdoc/html/render/write_shared/tests.rs" "/checkout/src/librustdoc/html/render/search_index.rs" "/checkout/src/librustdoc/html/render/write_shared.rs" "/checkout/src/librustdoc/html/render/context.rs" "/checkout/src/librustdoc/html/render/sorted_template/tests.rs" "/checkout/src/librustdoc/html/render/sidebar.rs" "/checkout/src/librustdoc/html/render/tests.rs" "/checkout/src/librustdoc/html/render/span_map.rs" "/checkout/src/librustdoc/html/render/sorted_template.rs" "/checkout/src/librustdoc/html/render/mod.rs" "/checkout/src/librustdoc/html/render/print_item.rs" "/checkout/src/librustdoc/html/toc/tests.rs" "/checkout/src/librustdoc/html/tests.rs" "/checkout/src/librustdoc/html/escape/tests.rs" "/checkout/src/librustdoc/html/escape.rs" "/checkout/src/librustdoc/html/highlight.rs" "/checkout/src/librustdoc/html/format.rs" "/checkout/src/librustdoc/html/highlight/fixtures/union.rs" "/checkout/src/librustdoc/html/highlight/fixtures/sample.rs" "/checkout/src/librustdoc/html/highlight/tests.rs" "/checkout/src/librustdoc/html/mod.rs" "/checkout/src/librustdoc/html/length_limit/tests.rs" "/checkout/src/librustdoc/scrape_examples.rs" "/checkout/src/librustdoc/doctest/rust.rs" "/checkout/src/librustdoc/doctest/markdown.rs" "/checkout/src/librustdoc/doctest/make.rs" "/checkout/src/librustdoc/doctest/tests.rs" "/checkout/src/librustdoc/doctest/runner.rs" "/checkout/src/librustdoc/visit_ast.rs" "/checkout/src/librustdoc/externalfiles.rs" "/checkout/src/librustdoc/fold.rs" "/checkout/src/librustdoc/docfs.rs" "/checkout/src/librustdoc/lib.rs" "/checkout/src/librustdoc/json/import_finder.rs" "/checkout/src/librustdoc/json/conversions.rs" "/checkout/src/librustdoc/json/mod.rs" "/checkout/src/librustdoc/core.rs" "/checkout/src/tools/nix-dev-shell/x/x.rs" "/checkout/src/tools/generate-copyright/src/main.rs" "/checkout/src/librustdoc/clean/auto_trait.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Tue Dec  3 21:36:45 UTC 2024
  network time: Tue, 03 Dec 2024 21:36:45 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/librustdoc/html/sources.rs
+++ b/src/librustdoc/html/sources.rs
@@ -50,7 +50,7 @@ struct LocalSourcesCollector<'a, 'tcx> {
     src_root: &'a Path,
 }
 
-fn is_real_and_local(span: clean::Span, sess: &Session) -> Option<RealFileName> {
+fn filename_real_and_local(span: clean::Span, sess: &Session) -> Option<RealFileName> {
     if span.cnum(sess) == LOCAL_CRATE
         && let FileName::Real(file) = span.filename(sess)
     {
@@ -66,7 +66,8 @@ impl LocalSourcesCollector<'_, '_> {
         let span = item.span(self.tcx);
         let Some(span) = span else { return };
         // skip all synthetic "files"
-        let Some(p) = is_real_and_local(span, sess).and_then(|file| file.into_local_path()) else {
+        let Some(p) = filename_real_and_local(span, sess).and_then(|file| file.into_local_path())
+        else {
             return;
         };
         if self.local_sources.contains_key(&*p) {
@@ -132,7 +133,7 @@ impl DocVisitor<'_> for SourceCollector<'_, '_> {
         // If we're not rendering sources, there's nothing to do.
         // If we're including source files, and we haven't seen this file yet,
         // then we need to render it out to the filesystem.
-        if let Some(filename) = is_real_and_local(span, sess) {
+        if let Some(filename) = filename_real_and_local(span, sess) {
             let span = span.inner();
             let pos = sess.source_map().lookup_source_file(span.lo());
             let file_span = span.with_lo(pos.start_pos).with_hi(pos.end_position());

